_posts/2023-02-07-announcing-data-prepper-2-1-0.md

```diff
@@ -0,0 +1,209 @@
+
+Data Prepper also now allows routing documents to specific OpenSearch shards by use of the existing `routing` parameter in OpenSearch. The new `routing_field` property on the OpenSearch sink will use properties from events to specify how to route within OpenSearch. Most users will prefer to allow OpenSearch to generate document ids and choose the shard from that. But, in some cases users need to specify these values and now Data Prepper offers users that option.
+
+Going hand-in-hand with the routing field, Data Prepper now allows pipeline authors to specify a complex field to use when specifying a document Id. Previously, a pipeline author would need to copy a value to a field in the root of the event to use it as a document ID.
+
+## Type conversion
+
+Data Prepper now supports a new [convert entry](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/mutate-event-processors#convertentryprocessor) processor as part of [mutate processors](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/mutate-event-processors#mutate-event-processors) to convert value from one type to another. This can be particularly useful in conditional expressions when using conditional routing.
+
+For example, in the following pipeline using this processor would allow a user to convert value of `status` key to `integer`. 
+
+```
+...
+processor:
+  - convert_entry_type:
+      key: "status"
+      type: "integer"
+...
+```
+
+## Other improvements
+
+In addition to the new features already described, Data Prepper 2.1.0 has a few other improvements. We want to highlight a few of them.
+
+* Data Prepper pipelines with OpenSearch sink now waits for OpenSearch to start during pipeline initialization if the sink is not available instead of failing.
+* OpenSearch sink now allows users to load index mapping and ISM template files from S3 bucket. Pipeline authors can configure it by providing S3 URI’s in existing configuration option to load them.
+* Data Prepper expressions now support `null` value keyword in conditions with equality operators.
+* To help with pipeline stability, Data Prepper now supports a circuit breaker on Java heap usage. This will cause the sources to stop accepting new data that could possibly cause the process to crash.
+* The documentation team has been updating the [Data Prepper documentation](https://opensearch.org/docs/latest/data-prepper/index/) by migrating many of the details from the GitHub repository into the primary documentation. 
+
+See the [release notes](https://github.com/opensearch-project/data-prepper/releases/tag/2.1.0) for a full list of changes.
+
+## Get started
+
+You can download Data Prepper or install a Docker container from the OpenSearch downloads page. The maintainers encourage all users to update to Data Prepper 2.1.0 to gain the improved stability, administration options, and feature set.
```

Comment:
Add link to the OpenSearch downloads page: https://opensearch.org/downloads.html